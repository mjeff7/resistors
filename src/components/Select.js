// @flow

import React from "react";

type Props<T> = {
  onChange: T => mixed,
  options: Array<T>,
  value?: T
};

export default ({ onChange, options, value }: Props<*>) => (
  <select onChange={e => onChange(e.target.value)} value={value}>
    {options.map((option, optionIndex) => (
      <option key={optionIndex} value={option}>
        {option}
      </option>
    ))}
  </select>
);
