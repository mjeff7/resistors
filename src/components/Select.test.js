// @flow

import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import Select from "./Select";

const pose = component => renderer.create(component).toJSON();

it("includes the options given to it", () => {
  const options = ["apple", "peppermint", "banana", "cinnamon", "cherry"];
  const component = <Select options={options} onChange={() => {}} />;

  expect(pose(component)).toMatchSnapshot();
});

it("calls onChange with value when anything is selected", () => {
  const options = ["a", "p", "b", "c"];
  const callback = jest.fn();
  const component = mount(<Select options={options} onChange={callback} />);

  component.find("select").simulate("change", { target: { value: "p" } });
  expect(callback).lastCalledWith("p");

  component.find("select").simulate("change", { target: { value: "a" } });
  expect(callback).lastCalledWith("a");

  expect(callback).toHaveBeenCalledTimes(2);
});

it("uses the passed in value if present", () => {
  const initialValue = "b";

  const options = ["a", "p", "b", "c"];
  const component = (
    <Select options={options} onChange={() => {}} value={initialValue} />
  );
  const mountedComponent = mount(component);

  const actualValue = mountedComponent.find("select").props().value;

  expect(actualValue).toBe(initialValue);
  expect(pose(component)).toMatchSnapshot();
});
