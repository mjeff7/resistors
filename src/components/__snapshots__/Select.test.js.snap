// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`includes the options given to it 1`] = `
<select
  onChange={[Function]}
  value={undefined}
>
  <option
    value="apple"
  >
    apple
  </option>
  <option
    value="peppermint"
  >
    peppermint
  </option>
  <option
    value="banana"
  >
    banana
  </option>
  <option
    value="cinnamon"
  >
    cinnamon
  </option>
  <option
    value="cherry"
  >
    cherry
  </option>
</select>
`;

exports[`uses the passed in value if present 1`] = `
<select
  onChange={[Function]}
  value="b"
>
  <option
    value="a"
  >
    a
  </option>
  <option
    value="p"
  >
    p
  </option>
  <option
    value="b"
  >
    b
  </option>
  <option
    value="c"
  >
    c
  </option>
</select>
`;
